src/main/java/org/terasology/multiBlock/recipe/LayeredMultiBlockFormItemRecipe.java

```diff
@@ -85,22 +88,21 @@ public boolean processActivation(ActivateEvent event) {
     }
 
     private boolean processDetectionForLayer(ActivateEvent event, int layerIndex, Vector3i basePosition) {
-        BlockEntityRegistry blockEntityRegistry = CoreRegistry.get(BlockEntityRegistry.class);
         LayerDefinition layerDefinition = layerDefinitions.get(layerIndex);
         Predicate<EntityRef> entityFilter = layerDefinition.entityFilter;
-        int minX = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.east()).x;
-        int maxX = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.west()).x;
-        int minZ = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.south()).z;
-        int maxZ = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.north()).z;
+        int minX = getLastMatchingInDirection( entityFilter, basePosition, Direction.RIGHT.asVector3i()).x;
+        int maxX = getLastMatchingInDirection( entityFilter, basePosition, Direction.LEFT.asVector3i()).x;
+        int minZ = getLastMatchingInDirection( entityFilter, basePosition, Direction.FORWARD.asVector3i()).z;
+        int maxZ = getLastMatchingInDirection( entityFilter, basePosition, Direction.BACKWARD.asVector3i()).z;
```

Comment:
`Vector3i.north` is defined as `(0, 0, 1)` while `Direction.BACKWARD` is `(0, 0, -1)`. Did you fix some bug here, or is this a mistake in the migration? 

```suggestion
        int minZ = getLastMatchingInDirection( entityFilter, basePosition, Direction.BACKWARD.asVector3i()).z;
        int maxZ = getLastMatchingInDirection( entityFilter, basePosition, Direction.FORWARD.asVector3i()).z;
```